### PR TITLE
added book entity and its repository, service

### PR DESCRIPTION
**Pull request by apanyutin**

### CODE REVIEW

File: src/main/java/com/example/service/impl/BookServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+public class BookServiceImpl implements BookService {
+    @Override
+    public Book save(Book book) {
+        return null;
+    }
+
+    @Override
+    public List findAll() {
+        return null;
+    }
```
Comment (okuzan) [inline src/main/java/com/example/service/impl/BookServiceImpl.java]: please add implementation

**Review by okuzan — CHANGES_REQUESTED** [review]

File: src/main/java/com/example/model/Book.java [hunk 2]
```diff
@@ -1,13 +1,27 @@
...
+import jakarta.persistence.Column;
 import jakarta.persistence.Entity;
 import jakarta.persistence.GeneratedValue;
 import jakarta.persistence.GenerationType;
 import jakarta.persistence.Id;
+import java.math.BigDecimal;
+import lombok.Data;
 
 @Entity
+@Data
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/model/Book.java]:
```suggestion
@Getter
@Setter
@Table(.....
```

File: src/main/java/com/example/repository/impl/BookRepositoryImpl.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+            session = sessionFactory.openSession();
+            transaction = session.beginTransaction();
+            session.persist(book);
+            transaction.commit();
+            return book;
+        } catch (Exception exception) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new RuntimeException("Can't save into DB the book: " + book, exception);
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/repository/impl/BookRepositoryImpl.java]: create custom exception

File: src/main/java/com/example/repository/impl/BookRepositoryImpl.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+            }
+        }
+    }
+
+    @Override
+    public List findAll() {
+        try (Session session = sessionFactory.openSession()) {
+            return session.createQuery("FROM Book", Book.class).getResultList();
+        } catch (Exception exception) {
+            throw new RuntimeException("Can't find all books in DB", exception);
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/repository/impl/BookRepositoryImpl.java]: same

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comments

File: src/main/java/com/example/model/Book.java (hunk 2 above)
Comment (apanyutin) [reply src/main/java/com/example/model/Book.java]: Why using of @Getter and @Setter are better than @Data?

Comment (apanyutin) [reply src/main/java/com/example/model/Book.java]: Performance?

Comment (Elena-Bruyako) [reply src/main/java/com/example/model/Book.java]: https://medium.com/@miguelangelperezdiaz444/the-hidden-dangers-of-lombok-annotations-in-your-java-code-what-you-need-to-know-8acdce2d6b89

File: src/main/java/com/example/repository/impl/BookRepositoryImpl.java [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+        try (Session session = sessionFactory.openSession()) {
+            return session.createQuery("FROM Book", Book.class).getResultList();
+        } catch (Exception exception) {
+            throw new DataProcessingException("Can't find all books in DB", exception);
+        }
+
+    }
+
+    @Override
+    public Book getById(Long id) {
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/repository/impl/BookRepositoryImpl.java]: It's out of scope of the task. 
But, if you would like to implement this method, use Optional<Book> as return object

File: src/main/java/com/example/repository/impl/BookRepositoryImpl.java [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+        try (Session session = sessionFactory.openSession()) {
+            return session.find(Book.class, id);
+        } catch (Exception exception) {
+            throw new EntityNotFoundException("Can't get book by id: " + id);
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/repository/impl/BookRepositoryImpl.java]:
```suggestion
            throw new DataProcessingException("Can't get book by id: " + id);
```

File: src/main/java/com/example/service/impl/BookServiceImpl.java [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+    @Override
+    public List<BookDto> getAll() {
+        return bookRepository.findAll().stream()
+                .map(bookMapper::toDto)
+                .toList();
+    }
+
+    @Override
+    public BookDto getById(Long id) {
+        Book book = bookRepository.getById(id);
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/service/impl/BookServiceImpl.java]:
```suggestion
        Book book = bookRepository.getById(id).orElseThrow(....
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comments

File: src/main/java/com/example/repository/impl/BookRepositoryImpl.java (hunk 5 above)
Comment (apanyutin) [reply src/main/java/com/example/repository/impl/BookRepositoryImpl.java]: From the video classes I've know that we have a agreement that get methods return Object and find methods return Optional. According to task we need implement get method https://i.imgur.com/AuHInVr.png . Do I need rename this method?

**Review by Elena-Bruyako — APPROVED** [review]
well done!

**Pull request merged**